pkg/controller/registry/resolver/resolver.go

```diff
@@ -55,6 +55,46 @@ func (w *debugWriter) Write(b []byte) (int, error) {
 	return n, nil
 }
 
+// DumpCache writes out the contents of the cache to logs
+// This is useful for debugging resolution errors
+func (r *Resolver) DumpCache(namespaces []string) {
```

Comment:
In addition to seeing what is in the cache, I would like to see what entries are returned by the lister, as it could be feeding stale entries into the cache.